torchvision/transforms/v2/functional/_utils.py

```diff
@@ -29,11 +29,12 @@ def _register_kernel_internal(dispatcher, input_type, *, datapoint_wrapper=True)
         raise ValueError(f"Dispatcher {dispatcher} already has a kernel registered for type {input_type}.")
 
     def decorator(kernel):
-        registry[input_type] = (
-            _kernel_datapoint_wrapper(kernel)
-            if issubclass(input_type, datapoints.Datapoint) and datapoint_wrapper
-            else kernel
-        )
+        # registry[input_type] = (
+        #     _kernel_datapoint_wrapper(kernel)
+        #     if issubclass(input_type, datapoints.Datapoint) and datapoint_wrapper
+        #     else kernel
+        # )
+        registry[input_type] = kernel
```

Comment:
We don't need to wrap/unwrap anymore.

Reply:
I think we still should. Although not technically needed for the functionality, this would be a step back in terms of performance. Since unwrapping no longer happens by default, all operations inside the kernels retain the type. In turn that means that the default `__torch_function__` logic is called over and over again:

```python
import torch
from torchvision import datapoints
from torchvision.transforms.v2 import functional as F


class MyImage(datapoints.Image):
    @classmethod
    def __torch_function__(cls, *args, **kwargs):
        global num_calls
        num_calls += 1
        return super().__torch_function__(*args, **kwargs)


num_calls = 0

image = MyImage(torch.rand(3, 256, 256))


print(num_calls)

F.adjust_hue(image, hue_factor=0.25)

print(num_calls)
```

```
0
68
```

I picked `adjust_hue` as worst offender (I think?) here to make my point clear. On `main` the same snippet prints

```
0
0
```

I remember that we spent quite some effort on removing these calls from the pipeline, since they piled up fast and in the end had a measurable impact on the performance. See for example https://github.com/pytorch/vision/pull/6807#issuecomment-1286652664.

Reply:
But you just defined a torchfunction on the sublass while this PR completely removes it. Are you sure that the "default torchfunction" is still invoked?

Either way I'm ok to bring back wrapping/unwrapping just to avoid any regression risk. It's an easy switch now that we have refactored registration. 

But I wonder whether the past results are still applicable to this PR

Reply:
> But you just defined a torchfunction on the sublass while this PR completely removes it. Are you sure that the "default torchfunction" is still invoked?

Yes. However, I didn't find a good way to put that into a snippet. You can just put a `print` there and see for yourself.

> But I wonder whether the past results are still applicable to this PR

In what sense? 

- The amount of `__torch_function__` calls? If so, then likely yes. The behavior you have in this PR is what we had before #6807.
- The impact on the runtime? I don't know. We would need to run some benchmarks to confirm. However, I don't see why it would have changed enough to make the impact insignificant.

Reply:
Hm yeah OK, I can reproduce the 68 calls to `Tensor.__torch_function__` even without overriding it. I'll put back the unwrapping to stay safe.

> The impact on the runtime?

Yeah, although looking more into it, it looks like the `Tensor.__torch_function__` is essentially the same as ours (even with the `DisableTorchFunctionSubclass` CM). I was hoping it'd be a lot simpler / faster, but it's unlikely to be the case.